tests/test_monitor_system.py

```diff
@@ -0,0 +1,126 @@
+import time
+from pathlib import Path
+
+from dvclive import Live
+from dvclive.monitor_system import MonitorCPU
+from dvclive.utils import parse_metrics
+
+
+def mock_psutil(mocker):
+    mocker.patch(
+        "dvclive.monitor_system.psutil.cpu_percent",
+        return_value=[10, 20, 30, 40, 50, 60],
+    )
+    mocker.patch("dvclive.monitor_system.psutil.cpu_count", return_value=6)
+
+    mocked_virtual_memory = mocker.MagicMock()
+    mocked_virtual_memory.percent = 20
+    mocked_virtual_memory.total = 4 * 1024**3
+
+    mocked_disk_usage = mocker.MagicMock()
+    mocked_disk_usage.used = 16 * 1024**3
+    mocked_disk_usage.percent = 50
+    mocked_disk_usage.total = 32 * 1024**3
+
+    mocking_dict = {
+        "virtual_memory": mocked_virtual_memory,
+        "disk_usage": mocked_disk_usage,
+    }
+    for function_name, return_value in mocking_dict.items():
+        mocker.patch(
+            f"dvclive.monitor_system.psutil.{function_name}",
+            return_value=return_value,
+        )
+
+
+def test_get_cpus_metrics_mocker(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        monitor = MonitorCPU(directories_to_monitor=["/", "/"])
+        monitor(live)
+        metrics = monitor._get_metrics()
+        monitor.end()
+
+    assert "system/cpu/usage (%)" in metrics
+    assert "system/cpu/count" in metrics
+    assert "system/cpu/parallelization (%)" in metrics
+    assert "system/ram/usage (%)" in metrics
+    assert "system/ram/usage (GB)" in metrics
+    assert "system/ram/total (GB)" in metrics
+    assert "system/disk/usage (%)/0" in metrics
+    assert "system/disk/usage (%)/1" in metrics
+    assert "system/disk/usage (GB)/0" in metrics
+    assert "system/disk/usage (GB)/1" in metrics
+    assert "system/disk/total (GB)/0" in metrics
+    assert "system/disk/total (GB)/1" in metrics
+
+
+def test_ignore_missing_directories(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        missing_directories = "______"
+        monitor = MonitorCPU(directories_to_monitor=["/", missing_directories])
+        monitor(live)
+        metrics = monitor._get_metrics()
+        monitor.end()
+
+    assert not Path(missing_directories).exists()
+
+    assert "system/disk/usage (%)/1" not in metrics
+    assert "system/disk/usage (GB)/1" not in metrics
+    assert "system/disk/total (GB)/1" not in metrics
+
+
+def test_monitor_system(tmp_dir, mocker):
```

Comment:
Still not that comfortable with this test. WDYT about splitting into 2 tests?

1. Call `Live(monitor_system=True)` and only test that it sets the monitor, calls it, and ends it.
2. Call `live = Live()` followed by `self._monitor_cpu = MonitorCPU(interval=0, num_samples=1)` and test that it logs all the expected values.

cc @iterative/dvc

Reply:
I agree on splitting the tests to test different functionalities. But the `MonitorCPU(interval=0, num_samples=1)` or any with a small interval value doesn't work if there is no sleep afterward. So I choose a small value to limit the time asleep.
Let me know if the current implementation seems better for you?

Reply:
Sounds good! It still seems like it could be flaky if the thread gets delayed, but not a blocker for me if it's not happening in practice. No need to duplicate the other discussion with @shcheklein, so let's resolve this and discuss there.